dask/dataframe/_pyarrow.py

```diff
@@ -76,5 +72,14 @@ def to_pyarrow_string(df):
     if (is_dataframe_like(df) or is_series_like(df)) and is_object_string_index(
         df.index
     ):
-        df.index = df.index.astype(pd.StringDtype("pyarrow"))
+        if isinstance(df.index, pd.MultiIndex):
+            for i, level in enumerate(df.index.levels):
+                if is_object_string_dtype(level.dtype):
+                    new_level = level.astype(pd.StringDtype("pyarrow"))
+                    # set verify_integrity=False to preserve index codes
+                    df.index = df.index.set_levels(
+                        new_level, level=i, verify_integrity=False
+                    )
```

Comment:
This looks functionally correct, but I'm wondering if these repeated `set_levels` calls result in lots of copies. Maybe @phofl knows what's happening under the hood

Reply:
To be clear, this isn't meant as a blocking comment. What you have here looks functionally correct. I'm just flagging this as something something that might be worth revisiting if this is making a bunch of copies and there's a reasonable way to reduce those copies.

Reply:
I hope @phofl can provide some advice.

Reply:
There are some validations happening under the hood that can create some overhead, but we don't copy the actual values and the most expensive validation happens with ``verify_integrity=True``, so this should not be too bad.

That said, I'd recommend collecting all changed levels as a ``list[Index]`` and then setting them at once in the end. This should remove as much overhead as possible.

Reply:
Thanks @phofl, that's very helpful.